### PR TITLE
Support Ubuntu 22.04

### PR DESCRIPTION
**Pull request by kenhys**
This draft PR is derived from https://github.com/fluent/fluentd/discussions/3704 and push a commit on https://github.com/fluent/fluent-package-builder/pull/378
to attempt to identify Ubuntu 22.04 buildability. (not ready for production)

* Use newer ruby only for Ubuntu 22.04 because of supporting OpenSSL 3.x

KNOWN ISSUE: dh_shlibdeps causes an error.

```
dpkg-shlibdeps: error: no dependency information found for /build/td-agent-4.3.1/debian/tmp/opt/td-agent/lib/libruby.so.3.1 (used by debian/
```

### CODE REVIEW

File: td-agent/Rakefile [hunk 1]
```diff
@@ -168,6 +168,21 @@ def tar_command
   end
 end
 
+def use_ruby3?
+  return false unless File.exist?("/etc/os-release")
+  os_release_entries = File.open("/etc/os-release").readlines.map(&:strip)
+  os_release_entries.include?("UBUNTU_CODENAME=jammy") and os_release_entries.include?("ID=ubuntu")
+end
```
Comment (cosmo0920) [inline td-agent/Rakefile:175]: I guess that it is not enough to switch using Ruby 3.1 or Ruby 2.7 on creating tarball and building.
We should bundle Ruby 3.1 and Ruby 2.7 tarballs on the stable source and switch them with this method.

Comment (ashie) [reply td-agent/Rakefile:175]: You are right, we don't want to depend on host OS for building GNU/Linux version of td-agent.

Comment (kenhys) [reply td-agent/Rakefile:178]: Changed to bundle both of versions.
Thanks.

**Comment by kenhys** [comment]
```
% tar tvf td-agent-4.3.1.tar.gz |\grep ruby-  
...
-rw-r--r-- kenhys/docker 16919639 2022-04-28 16:34 td-agent-4.3.1/td-agent/downloads/ruby-2.7.6.tar.gz
-rw-r--r-- kenhys/docker 20553628 2022-04-28 16:34 td-agent-4.3.1/td-agent/downloads/ruby-3.1.2.tar.gz
```

File: td-agent/Rakefile [hunk 2]
```diff
@@ -168,11 +168,28 @@ def tar_command
   end
 end
 
+def use_ruby3?
+  # Check whether ubuntu 22.04 (jammy) or not
+  # Basically, TD Agent adopts ruby 2.7 as default version, but on Ubuntu 22.04,
+  # use ruby 3.x because distribution switches to OpenSSL 3.x.
+  return false unless File.exist?("/etc/os-release")
+  os_release_entries = File.open("/etc/os-release").readlines.map(&:strip)
+  os_release_entries.include?("UBUNTU_CODENAME=jammy") and os_release_entries.include?("ID=ubuntu")
```
Comment (cosmo0920) [inline td-agent/Rakefile:177]: Nitpick: This is not required but also CentOS Stream 9 (and RockyLinux 9) will be needed to switch using Ruby 3.1.
This is just for you information but we will need to handle this Ruby 3.1 switching in the future.

**Comment by cosmo0920** [comment]
FYI: aptly cannot handle Zstandard compressed control.tar.zst and data.tar.zst within debian package. This should be affected for Ubuntu Jammy (22.04) packages.

**Comment by kenhys** [comment]
> FYI: aptly cannot handle Zstandard compressed control.tar.zst and data.tar.zst within debian package, now. This should be affected for Ubuntu Jammy (22.04) packages. [aptly-dev/aptly#1050](https://github.com/aptly-dev/aptly/pull/1050)

Thanks, as you pointed out, aptly 1.4.0+ds1-6 doesn't support it.

**Comment by kenhys** [comment]
As a workaround, we can use xz as default package compression again.

```
Local repo [td-agent4-jammy] successfully added.
You can run 'aptly repo add td-agent4-jammy ...' to add packages to repository.
Loading packages...
[+] td-agent_4.3.1-1_amd64 added

Snapshot td-agent4-jammy-4.3.1-1 successfully created.
You can run 'aptly publish snapshot td-agent4-jammy-4.3.1-1' to publish snapshot as Debian repository.
Loading packages...
Generating metadata files and linking package files...
Finalizing metadata files...
Signing file 'Release' with gpg, please enter your passphrase when prompted:
Clearsigning file 'Release' with gpg, please enter your passphrase when prompted:

Snapshot td-agent4-jammy-4.3.1-1 has been successfully published.
Please setup your webserver to serve directory '/work/fluentd/fluent-package-builder/.aptly/public' with autoindexing.
Now you can add following line to apt sources:
  deb http://your-server/ubuntu/jammy/ jammy contrib
Don't forget to add your GPG key to apt with apt-key.

You can also use `aptly serve` to publish your repositories over HTTP quickly.

Don't forget to sync from /work/fluentd/fluent-package-builder/.aptly/.public/ to td-agent-release/4/

```

**Review by ashie — APPROVED** [review]
LGTM

**Comment by ashie** [comment]
Thanks!
We'll release it as an experimental package.

**Pull request merged**